wandb/proto/wandb_api.proto

```diff
@@ -2,42 +2,75 @@ syntax = "proto3";
 
 package wandb_internal;
 
-import "wandb/proto/wandb_base.proto";
+import "wandb/proto/wandb_settings.proto";
 
 option go_package = "core/pkg/service_go_proto";
 
 // ApiRequest is a request to the backend process
 // to perform an action related to an API call.
 message ApiRequest {
   oneof request {
-    ReadRunHistoryApiRequest read_run_history = 1;
+    ReadRunHistoryRequest read_run_history_request = 1;
   }
 }
 
 // ApiResponse is a response from the backend process for an ApiRequest.
 message ApiResponse {
   oneof response {
-    ReadRunHistoryApiResponse read_run_history = 1;
+    ReadRunHistoryResponse read_run_history_response = 1;
+    ApiErrorResponse api_error_response = 2;
   }
 }
 
-// Start of Read run history
-// ReadRunHistoryApiRequest is a request to get a slice of a run's history.
+message ApiErrorResponse {
+  string message = 1;
+}
+
+// Start of Scan run history
+
+message ReadRunHistoryRequest {
+  oneof request {
+    ScanRunHistoryInit scan_run_history_init = 1;
+    ScanRunHistory scan_run_history = 2;
+    ScanRunHistoryCleanup scan_run_history_cleanup = 3;
+  }
+}
 
-message ReadRunHistoryApiRequest {
+message ReadRunHistoryResponse {
+  oneof response {
+    ScanRunHistoryInitResponse scan_run_history_init = 1;
+    RunHistoryResponse run_history_response = 2;
+  }
+}
+
+// ScanRunHistoryInit is a request to initialize
+// a scan over a run's history.
+//
+// Because scan history request will be made over a series of requests,
+// We want to keep initialize and keep track of the resources
+// associated with the scan request.
+message ScanRunHistoryInit {
   string entity = 1;
   string project = 2;
   string run_id = 3;
   repeated string keys = 4;
-  int64 min_step = 5;
-  int64 max_step = 6;
+  Settings settings = 5;
```

Comment:
Do you really need the whole Settings thing every time you initialize a history scan? also, couldn't you use the settings from the next PR on the stack?